src/csharp/Grpc.Tools/Common.cs

```diff
@@ -60,7 +60,7 @@ static Platform()
                : RuntimeInformation.IsOSPlatform(OSPlatform.OSX) ? OsKind.MacOsX
                : OsKind.Unknown;
 
-            switch (RuntimeInformation.OSArchitecture)
+            switch (RuntimeInformation.ProcessArchitecture)
```

Comment:
I think this is the way to go! I noted that in another comment somewhere, in another conversation with you, but cannot find it. If we are an N-bit process, then the system is certainly capable of starting N-bit processes, and the value is used for nothing else in the tooling except locating the correct executable. Even if we're on Mac, where you noticed mono was 32-bit by default, we can run 32-bit protoc.